presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -148,34 +155,48 @@
 public class ThriftHiveMetastore
         implements HiveMetastore
 {
+    private static final Logger log = Logger.get(ThriftHiveMetastore.class);
+
+    private static final String DEFAULT_METASTORE_USER = "presto";
+
     private final ThriftHiveMetastoreStats stats;
     private final HiveCluster clientProvider;
     private final Function<Exception, Exception> exceptionMapper;
+    private final HdfsEnvironment hdfsEnvironment;
+    private final HdfsContext hdfsContext;
     private final boolean impersonationEnabled;
     private final boolean isMetastoreAuthenticationEnabled;
+    private final boolean deleteFilesOnTableDrop;
 
     @Inject
-    public ThriftHiveMetastore(HiveCluster hiveCluster, MetastoreClientConfig config)
+    public ThriftHiveMetastore(HiveCluster hiveCluster, MetastoreClientConfig config, HdfsEnvironment hdfsEnvironment)
     {
         this(
                 hiveCluster,
                 new ThriftHiveMetastoreStats(),
                 identity(),
+                hdfsEnvironment,
                 requireNonNull(config, "config is null").isMetastoreImpersonationEnabled(),
+                requireNonNull(config, "config is null").isDeleteFilesOnTableDrop(),
                 requireNonNull(config, "config is null").getHiveMetastoreAuthenticationType() != HiveMetastoreAuthenticationType.NONE);
     }
 
     public ThriftHiveMetastore(
             HiveCluster hiveCluster,
             ThriftHiveMetastoreStats stats,
             Function<Exception, Exception> exceptionMapper,
+            HdfsEnvironment hdfsEnvironment,
             boolean impersonationEnabled,
+            boolean isDeleteFilesOnTableDrop,
             boolean isMetastoreAuthenticationEnabled)
     {
+        this.hdfsContext = new HdfsContext(new ConnectorIdentity(DEFAULT_METASTORE_USER, Optional.empty(), Optional.empty()));
         this.clientProvider = requireNonNull(hiveCluster, "hiveCluster is null");
+        this.hdfsEnvironment = requireNonNull(hdfsEnvironment, "hdfsEnvironment is null");
         this.stats = requireNonNull(stats, "stats is null");
         this.exceptionMapper = requireNonNull(exceptionMapper, "exceptionMapper is null");
         this.impersonationEnabled = impersonationEnabled;
+        this.deleteFilesOnTableDrop = isDeleteFilesOnTableDrop;
```

Comment:
lets use `deleteFilesOnTableDrop` everywhere for variable names.

Reply:
Done